integration/rpctest/rpc_harness.go

```diff
@@ -563,6 +563,99 @@ func NextAvailablePort() int {
 	panic("no ports available for listening")
 }
 
+// NextAvailablePortForProcess returns the first port that is available for
+// listening by a new node, using a lock file to make sure concurrent access for
+// parallel tasks within the same process don't re-use the same port. It panics
+// if no port is found and the maximum available TCP port is reached.
+func NextAvailablePortForProcess(pid int) int {
+	lockFile := fmt.Sprintf(
+		"%s/rpctest-port-pid-%d.lock", os.TempDir(), pid,
```

Comment:
curious about whether this would behave the same on windows?

Reply:
Oh, good point. I think we need to explicitly close a file before we can delete it. And we should probably use `filepath.Join()` instead of `fmt.Sprintf()` to create the path. Will fix.

Reply:
Updated things to be more Windows friendly.